components/PyTorch/_samples/Train_fully-connected_network.pipeline.py

```diff
@@ -0,0 +1,56 @@
+from kfp import components
+
+
+chicago_taxi_dataset_op = components.load_component_from_url('https://raw.githubusercontent.com/kubeflow/pipelines/e3337b8bdcd63636934954e592d4b32c95b49129/components/datasets/Chicago%20Taxi/component.yaml')
+pandas_transform_csv_op = components.load_component_from_url('https://raw.githubusercontent.com/kubeflow/pipelines/6162d55998b176b50267d351241100bb0ee715bc/components/pandas/Transform_DataFrame/in_CSV_format/component.yaml')
+
+create_fully_connected_pytorch_network_op = components.load_component_from_url('https://raw.githubusercontent.com/kubeflow/pipelines/4e1facea1a270535b515a9e8cc59422d1ad76a9e/components/PyTorch/Create_fully_connected_network/component.yaml')
+train_pytorch_model_from_csv_op = components.load_component_from_url('https://raw.githubusercontent.com/kubeflow/pipelines/603342c4b88fe2d69ff07682f702cd3601e883bb/components/PyTorch/Train_PyTorch_model/from_CSV/component.yaml')
+convert_to_onnx_from_pytorch_script_module_op = components.load_component_from_url('https://raw.githubusercontent.com/kubeflow/pipelines/e011e4affa85542ef2b24d63fdac27f8d939bbee/components/PyTorch/Convert_to_OnnxModel_from_PyTorchScriptModule/component.yaml')
+
+
+def pytorch_pipeline():
+    feature_columns = ['trip_seconds', 'trip_miles', 'pickup_community_area', 'dropoff_community_area', 'fare', 'tolls', 'extras']  # Excluded 'trip_total'
+    label_column = 'tips'
+    network = create_fully_connected_pytorch_network_op(
+        layer_sizes=[len(feature_columns), 100, 10, 1],
+        activation_name='elu',
+    ).output
+
+    training_data = chicago_taxi_dataset_op(
+        where='trip_start_timestamp >= "2019-01-01" AND trip_start_timestamp < "2019-02-01"',
+        select=','.join([label_column] + feature_columns),
+        limit=10000,
+    ).output
+
+    training_data = pandas_transform_csv_op(
+        table=training_data,
+        transform_code='''df = df.fillna({'tolls': 0.0, 'extras': 0.0}); df = df.dropna(axis='index')''',
+    ).output
+
+    trained_model = train_pytorch_model_from_csv_op(
+        model=network,
+        training_data=training_data,
+        label_column_name=label_column,
+        loss_function_name='mse_loss',
+        # Optional:
+        batch_size=32,
+        number_of_epochs=2,
+        random_seed=0,
+        learning_rate=0.1,
+        optimizer_name='Adadelta',
+        optimizer_parameters={},
+    ).outputs['trained_model']
+
+    convert_to_onnx_from_pytorch_script_module_op(
```

Comment:
`converted_model_path` is not exposed from this component, am I missing something on how to provide model file path?

Reply:
If I understand correctly, you are asking how to provide argument for the `converted_model_path` parameter of the original `convert_to_onnx_from_pytorch_script_module` function. Is that correct?

The `converted_model_path` parameter is exposed as the `converted_model` *output* of the component.
Paths for all component outputs are auto-generated by the system. The `converted_model_path` will be set to something like `/tmp/outputs/converted_model/data` at runtime (and even a bit at compile-time). There is no reason for the pipeline author to want to control the path, since the path is container-local and disappears right after the component finishes.

The pipeline author sets up the data connections (by passing outputs to inputs), but does not usually control how the data gets from one component to another, which is implementation detail.

If the pipeline author wants to export the model to a specific place, for example GCS, they need to do that explicitly and pass the `converted_model` output to the `upload_to_gcs` component.

Reply:
I see, thank you Alexey for the explaination! It makes sense for model path being auto-generated and used in another componet.